src/components/popup/details/purposes/purposes.jsx

```diff
@@ -49,12 +51,18 @@ export default class Purposes extends Component {
 		if (selectedPurpose) {
 			if (selectedPurpose.custom) {
 				selectCustomPurpose(selectedPurpose.id, isSelected);
+			} else if (isPublisher) {
+				selectStandardPurpose(selectedPurpose.id, isSelected);
 			} else {
 				selectPurpose(selectedPurpose.id, isSelected);
 			}
 		}
 	};
 
+	createHandleSelectPurpose = (isPublisher) => {
+		return (data) => this.handleSelectPurpose(data, isPublisher);
+	};
+
```

Comment:
Could be even shorter:

```suggestion
createHandleSelectPurpose = (isPublisher) => (data) => this.handleSelectPurpose(data, isPublisher);
```

Reply:
But I'm not sure if it would be more readable ;)